core/src/main/java/org/elasticsearch/index/mapper/internal/ParentFieldMapper.java

```diff
@@ -222,25 +238,43 @@ public Query termsQuery(List values, @Nullable QueryParseContext context) {
         }
     }
 
-    private final String type;
+    private final String parentType;
+    // determines the field data settings
+    private MappedFieldType childJoinFieldType;
+    // has no impact of field data settings, is just here for creating a join field, the parent field mapper in the child type pointing to this type determines the field data settings for this join field
+    private final MappedFieldType parentJoinFieldType;
+    private final MappedFieldType defaultJoinFieldType;
```

Comment:
Why do we need this default as a member? Isn't it always Defaults.JOIN_FIELD_TYPE?